scripts/gulpfiles/docs_tasks.js

```diff
@@ -0,0 +1,129 @@
+const { execSync } = require('child_process');
+const { Extractor } = require('markdown-tables-to-json');
+const fs = require('fs');
+const gulp = require('gulp');
+const header = require('gulp-header');
+const replace = require('gulp-replace');
+
+const docsDir = 'docs';
+
+/**
+ * Run API Extractor to generate the intermediate json file.
+ */
+const generateApiJson = function(done) {
+    execSync('api-extractor run --local', { stdio: 'inherit' });
+    done();
+}
+
+/**
+ * API Extractor output spuriously renames some functions. Undo that.
+ * See https://github.com/microsoft/rushstack/issues/2534
+ */
+const removeRenames = function() {
+    return gulp.src('temp/blockly.api.json')
+        .pipe(replace('_2', ''))
+        .pipe(gulp.dest('temp/'));
```

Comment:
Maybe, but since you're in the middle of reconfiguring the build pipeline, I didn't want to mess with it. Do you copy everything from build to somewhere else at any point?

Reply:
> Maybe, but since you're in the middle of reconfiguring the build pipeline, I didn't want to mess with it.

Reasonable enough.

> Do you copy everything from build to somewhere else at any point?

In the revised pipeline there is essentially _no_ copying; things are written directly to `dist/` if that's where they need to end up.  For now, though, there _is_ some copying and so you are wise to avoid putting anything new into `build/` lest it get copied inadvertently.